extensions/advanced_sequencing/main.md

```diff
@@ -0,0 +1,34 @@
+# DIDComm Advanced Sequencing Extension
+
+## Summary 
+
+In a simple interaction between two parties that consists of a request and a response, the message types usually imply the order without ambiguity. However, some protocols have more complex requirements. For example, an auction protocol that consists of `bid` messages might produce different outcomes depending on whether a lower bid arrives before or after a higher bid. When such protocols unfold over asynchronous transports that don't guarantee a delivery order, and when more than two participants are involved, understanding the intended sequence requires extra data.
+
+## Scope
+
+This extension does not specify how gaps or out-of-order messages should be *handled* -- only how they should be *detected*. That is because some protocols may consider out-of-order messages harmless, whereas other protocols may be highly sensitive to problems triggered by such conditions. Thus, the scope of this extension is *detecting sequencing issues*. Any protocol that cares about sequencing issues SHOULD use this mechanism as its foundational semantics for understanding the problem, and MAY choose to react to the issues in whatever way the protocol designers deem appropriate. 
+
+## Reference
+
+DIDComm messages that support this extension offer three headers related to sequencing: `sender_order`, `sent_count`, and `received_orders`. These headers are only relevant to certain protocols. Protocol definitions MUST declare whether they use any of these headers. This is because general implementations of DIDComm are not required to implement support for the headers unless/until they intend to support protocols that use them.
+
+### `sender_order`
```

Comment:
Regarding `sender_order` and `sent_count` in concept are interesting and behave similarly to a TTL for cellular networks. However, one of the interesting things to note is that the trust the cellular network places on their devices (which they don't control) have allowed for an interesting vulnerability described in this [video](https://www.youtube.com/watch?v=alMsu_tfZPE).

My take away from this is that while `sender_order` and `sent_count` are useful, they couldn't be trusted by the recipient. For example, Mallory could start by sending `sender_order=10` and `sent_count=4` in an effort to disrupt thing so we should make sure that the protocol designers are aware of this limitation.

Reply:
I agree that a malicious *sender* could send bad data. Nobody else could alter what the sender declares, however -- since these headers are in the plaintext that gets encrypted.

I agree that we should write up a security note about this. However, this vulnerability is not really very specific to this extension; it exists with the send time in core headers as well.

Reply:
+1 to security note for this. In most cases, this is a useful feature and really all we're trying to achieve is make note that the protocol designers need to make note of these things in their security considerations section.